### ISSUE TITLE
why use x_max while computing softmax in function log_sum_exp()?

### ISSUE THREAD

**Issue by huabinhuang1994** [amdegroot/ssd.pytorch#203]
def log_sum_exp(x)    
    x_max = x.data.max()
    return torch.log(torch.sum(torch.exp(x-x_max), 1, keepdim=True)) + x_max
In this function ,if we remove x_max,the output of this function is just the same,so why should we use the x_max ?

**Comment by huabinhuang1994** [comment]
Thank you, got it.
